std/range/package.d

```diff
@@ -4196,6 +4209,12 @@ auto iota(E)(E end)
 // Specialization for floating-point types
 auto iota(B, E, S)(B begin, E end, S step)
 if (isFloatingPoint!(CommonType!(B, E, S)))
+in
+{
+    assert(step != 0, "iota: step must not be 0");
+    assert((end - begin) / step >= 0, "iota: incorrect startup parameters");
```

Comment:
I just took them from (old) lines 4213 and 4215.

iota needs a little t.l.c. w.r.t. numerical concerns, but that's outside the scope of this pull.